.github/workflows/deploy_docs.yml

```diff
@@ -0,0 +1,45 @@
+
+name: deploy docs
+on:
+  release:
+    types: [published]
+    
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: install prerequisites
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y \
+          libdrm-dev \
+          libegl1-mesa-dev \
+          libgl1-mesa-dev \
+          libx11-dev \
+          libxext-dev \
+          libxfixes-dev \
+          libwayland-dev \
+          libx11-xcb-dev \
+           libxcb-dri3-0 \
+          doxygen 
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.10' 
+
+    - name: Build
+      run: |
+        python -m pip install ninja
```

Comment:
why install via pip vs. from ubuntu packages?

Reply:
Ninja is not a python module so the above does not make sense. Ideally both ninja and meson (below) should be installed with "apt-get install ..." if needed